File: Primtall.js

function faktoriser(tall) {
    primtall = hentPrimtall(tall + 1); //Henter alle primtall til og med tall mottat (kan gjøres mer effektivt)
    faktorer = [];
    ikkeEn = tall != 1;
    if (primtall.length > 0) {
        while (ikkeEn) {
            for (var i = 0; i < primtall.length; i++) {
                if (tall % primtall[i] == 0) {
                    faktorer.push(primtall[i]);
                    tall /= primtall[i];
                    ikkeEn = tall != 1;
                    break;
                }
            }
        }
    }

    return factors
}

function hentPrimtall(tall) {
    liste = [2];
    if (tall > 2) {
        for (var i = 2; i < tall; i++) {
            erPrimtall = true;
            for (var j = 0; j < liste.length; j++) {
                if (i % liste[j] == 0) {
                    erPrimtall = false;
                    break;
                }
            }
            if (erPrimtall) {
                liste.push(i);
            }
        }
    }
    return liste;
}

File: Primtall.py
import math

num = input("Hvilket tall skal sjekkes? \n")

#Factorizes a number
def faktoriser(tall):
    primtall = hentPrimtall(math.ceil(tall / 2))
    faktorer = []
    ikkeEn = tall != 1
    if len(primtall) > 0:
        while ikkeEn:   #While number hasn't reached 1
            ikkeEndret = True
            for verdi in primtall:
                if tall % verdi == 0:
                    faktorer.append(verdi)
                    tall /= verdi
                    ikkeEn = tall != 1
                    ikkeEndret = False
                    break
            if ikkeEndret:
                faktorer.append(int(tall))
                ikkeEn = False
    return faktorer


def hentPrimtall(tall):
    liste = [2]
    if tall > 2:
        for i in range(2,tall):
            erPrimtall = True
            for primtall in liste:
                if i % primtall == 0:
                    erPrimtall = False
                    break
            if erPrimtall:
                liste.append(i)

    return liste


def main(tall):
    try:
        print(faktoriser(int(tall)))
    except Exception as e:
        print(e)


main(num)
